File: src/test/java/com/github/kayjamlang/tests/containers/classes/AnonymousObjectContainerTest.java
package com.github.kayjamlang.tests.containers.classes;

import com.github.kayjamlang.core.Stmt;
import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.containers.ObjectContainer;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.*;

public class AnonymousObjectContainerTest {

    private static KayJamParser parser;

    @BeforeClass
    public static void prepare(){
        parser = new KayJamParser(new KayJamLexer("object{var test = 123;fun test(){return test}}"));
    }

    @Test
    public void test() throws Exception {
        Stmt stmt = parser.readStmt();

        assertNotNull(stmt);
        assertSame(ObjectContainer.class, stmt.getClass());

        ObjectContainer objectContainer = (ObjectContainer) stmt;
        assertEquals(1, objectContainer.variables.size());
        assertEquals(1, objectContainer.functions.size());
    }
}

File: src/test/scala/com/github/kayjamlang/tests/containers/classes/AnonymousObjectContainerTest.scala
package com.github.kayjamlang.tests.containers.classes

import com.github.kayjamlang.core.{KayJamLexer, KayJamParser}
import com.github.kayjamlang.core.containers.ObjectContainer
import org.junit.Assert._
import org.junit.{BeforeClass, Test}


object AnonymousObjectContainerTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("object{var test = 123;fun test(){return test}}"))
  }
}

class AnonymousObjectContainerTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = AnonymousObjectContainerTest.parser.readTopExpression
    assertNotNull(expression)
    assertSame(classOf[ObjectContainer], expression getClass)
    val objectContainer = expression.asInstanceOf[ObjectContainer]
    assertEquals(1, objectContainer.variables size)
    assertEquals(1, objectContainer.functions size)
  }
}